sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementSharesExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Storage.Files.Shares.Models;
+
+namespace Azure.Storage.DataMovement.Files.Shares
+{
+    internal static partial class DataMovementSharesExtensions
+    {
+        internal static ShareFileUploadOptions ToShareFileUploadOptions(
+            this ShareFileStorageResourceOptions options)
+            => new()
+            {
+                Conditions = options?.DestinationConditions,
+                TransferValidation = options?.UploadTransferValidationOptions,
+            };
+
+        internal static ShareFileUploadRangeOptions ToShareFileUploadRangeOptions(
+            this ShareFileStorageResourceOptions options)
+            => new()
+            {
+                Conditions = options?.DestinationConditions,
+                TransferValidation = options?.UploadTransferValidationOptions,
+            };
+
+        internal static ShareFileUploadRangeFromUriOptions ToShareFileUploadRangeFromUriOptions(
+            this ShareFileStorageResourceOptions options)
+            => new()
+            {
+                Conditions = options?.DestinationConditions,
+            };
+
+        internal static StorageResourceProperties ToStorageResourceProperties(
+            this ShareFileProperties properties)
+            => new(
```

Comment:
This is probably out of scope for this PR, but I think we need to think about `StorageResourceProperties` and what we actually need from it in the base package and how we can get sub-package specific stuff out. Right now, this has a bunch of blob specific stuff and now bringing it into Files makes it more obvious that that is an issue.

Reply:
I agree. I was thinking the same thing when I was writing this.